### PR TITLE
[DEV-976] extend defi tx

### PR DESCRIPTION
**Pull request by Krysh90**

### CODE REVIEW

File: gateway/local-main.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+  async run(): Promise<void> {
+    let ownerWallet: string | undefined = undefined;
+    do {
+      ownerWallet = await this.setupNeededComponents();
+      if (!ownerWallet) {
+        this.logger.info('... retrying in 5 seconds ...');
+        await Util.sleep(5);
+      }
+    } while (!ownerWallet);
```
Comment (davidleomay) [inline gateway/local-main.ts:28]: The retry got removed in the productive main. Do you want to keep it here?

File: shared/transaction/validator.ts [hunk 2]
```diff
@@ -21,7 +30,16 @@ export class Validator {
...
+      Validator.sendToLiq(tx, liqScript) ||
+      Validator.sendAccountToAccount(tx) ||
+      Validator.createVault(tx, script) ||
+      Validator.depositToVault(tx, script) ||
+      Validator.withdrawFromVault(tx, script) ||
+      Validator.takeLoan(tx, script) ||
+      Validator.paybackLoan(tx, script) ||
+      Validator.addPoolLiquidity(tx, script) ||
+      Validator.removePoolLiquidity(tx, script) ||
+      Validator.compositeSwap(tx, script)
```
Comment (davidleomay) [inline shared/transaction/validator.ts:42]: Looks repetitive 🤣  what about a list of allowed DeFi transactions and check them in a method (e.g. Validator.deFiTransaction) with a for loop?

Comment (Krysh90) [reply shared/transaction/validator.ts:42]: It needs to be changed anyway. There is an own ticket for it. To improve this mechanism. As yes it is suboptimal. I mean performance wise has really low impact, but it looks ugly

File: gateway/local-main.ts (hunk 1 above)
Comment (Krysh90) [reply gateway/local-main.ts:28]: yes it is nice for testing. as API is restarting or is not running yet while restarting all components

Comment (davidleomay) [reply gateway/local-main.ts:28]: Ok 👍

**Review by davidleomay — APPROVED** [review]
Improve validator in future PR.

**Pull request merged**